### DISCUSSION TITLE
[Question] - Smallest polygon from set of lines surrounding a point

### DISCUSSION THREAD

**Discussion by CrommVardek** [Turfjs/turf#2339] [Q&A]
Hello, 

This is not an issue, but rather a question: Using Turf, is there an "easy" way to find the smallest polygon, from a set of lines, in which a given point is inside that polygon ?

I've found solutions (without the use of Turf) for that problem, Turf would help implementing such solutions (with some of its functions such as area and pointToLineDistance), so this is nice. But is there a Turf function that would implement a solution for that problem ? I've searched the doc but couldn't find a function (or a set of function) that would solve this.

Thanks!

**Comment by rowanwins** [comment]
Currently Turf doesn't have anything to support that. 
And I can't say I've seen anything similar in the broader spatial analysis ecosystem - are you able to provide any references @CrommVardek? For examples do the lines touch or is the algorithm expected to compute a concave or convex hull around some candidates? Also what is the use case?

Cheers

**Reply by CrommVardek** [reply]
I don't have any reference in that regard. I found an algorithm (on stack-overlow, that needed some adaptations) that solves our issue, but in the mean-time we also found another way to solve this problem using turf (which is better and faster).

Yes, the lines touch, they are in a finite space, the space in which the candidate point is in is delimited by 4 lines that form polygon, so at worst, there is that polygon. The use case is not easy to explain, but bascially it's based on a country's law for physical entities of a particular sector that would what to move their physical entitie(s). 

Back to the solution we used (using Turf) : We compute all the potential polygons where the candidate point is in (we have an easy way to do this, because lines are created in a certain way), with all the polygons, we iterate over them, two by two, taking their intersection, the result, if it is a polygon (and with an area) is a new candidate polygon. We iterate over all polygons, eliminating those that have a greater area than the result. We can do this while being sure that our candidate point is still in the intersection of two polygons because of some properties of our use-case (I can't really go into too much details). After the iteration we do end up with the smallest polygon from a set of lines (which are parts of a set of lines) that surrounds a point. 

Sorry I cannot give you too much details. This solution works for us because we do have some properties applying on our use-case, which might not be the case for more generic use-cases.

**Comment by smallsaucepan** [comment]
Hasn't been much discussion on this for some time, so will cloase. If still a current need or new info is available please reopen.
